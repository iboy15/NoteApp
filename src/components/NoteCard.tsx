import { StyleSheet, TouchableOpacity } from 'react-native'
import { Text, View } from './Themed'

import { getParsedDate, height, width } from '../utils'
import { FontAwesome } from '@expo/vector-icons'
import useColorScheme from '../hooks/useColorScheme'

export default function NoteCard({
  onPress,
  body,
  title,
  color,
  created_at,
  is_favorite,
  setAsFavorite,
  archived,
}) {
  const colorScheme = useColorScheme()

  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.listItem, { backgroundColor: color }]}
    >
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.body} numberOfLines={6}>
        {body}
      </Text>
      <View style={styles.bottomContainer}>

          <Text style={{ fontStyle: 'italic',top:7 }}>
            {getParsedDate(created_at)} {'\n'}
          </Text>
    
        <TouchableOpacity
          onPress={setAsFavorite}
          style={[
            styles.favoriteBtn,
            {
              backgroundColor: colorScheme === 'dark' ? 'black' : 'white',
            },
          ]}
        >
          <FontAwesome
            name='heart'
            size={15}
            color={
              is_favorite
                ? '#d64161'
                : colorScheme === 'dark'
                ? 'white'
                : 'black'
            }
          />
        </TouchableOpacity>
        
      </View>
      {  archived &&  <View style={{flexDirection:'row',backgroundColor:'transparent',alignItems:'center'}}>
      <FontAwesome
            name='trash'
            size={15}
            color={
              colorScheme === 'dark'
                ? 'white'
                : 'black'
            }
          />
             <Text style={{ fontWeight: 'bold',fontSize:14,marginLeft:5}}>Archived</Text>
      </View>}
  
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    paddingTop: 20,
  },
  title: { fontWeight: 'bold', marginBottom: 5 },
  listItem: {
    width: width / 2 - 30,
    margin: 10,
    borderRadius: 10,
    overflow: 'hidden',
    padding: 10,
  },
  bottomContainer: {
    flexDirection: 'row',
    backgroundColor: 'transparent',
    marginTop: 'auto',
    alignItems:'center',
  },
  favoriteBtn: {
    height: 28,
    width: 28,
    borderRadius: 19,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 'auto',
    paddingLeft: 1,
  },
  body: { marginBottom: 10 },
})
