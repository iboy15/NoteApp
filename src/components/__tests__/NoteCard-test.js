import * as React from 'react';
import renderer from 'react-test-renderer';
import NoteCard from '../NoteCard'

it(`renders correctly`, async () => {
  const tree = await renderer.create(<NoteCard onPress={()=>{}}
    body='test'
    title='test'
    color='red'
    created_at={new Date()}
    is_favorit={true}
    setAsFavorite={()=>{}}
    archived={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});
