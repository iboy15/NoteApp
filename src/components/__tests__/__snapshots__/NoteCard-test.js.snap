// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "borderRadius": 10,
      "margin": 10,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 10,
      "width": 345,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontWeight": "bold",
          "marginBottom": 5,
        },
      ]
    }
  >
    test
  </Text>
  <Text
    numberOfLines={6}
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
  >
    test
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "marginTop": "auto",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontStyle": "italic",
            "top": 7,
          },
        ]
      }
    >
      NaN-NaN-NaN
       
      

    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 19,
          "height": 28,
          "justifyContent": "center",
          "marginLeft": "auto",
          "opacity": 1,
          "paddingLeft": 1,
          "width": 28,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
